pyquil/gate_matrices.py

```diff
@@ -0,0 +1,292 @@
+                  [0, 0, 0, 0, 1, 0, 0, 0],
+                  [0, 0, 0, 0, 0, 0, 1, 0],
+                  [0, 0, 0, 0, 0, 1, 0, 0],
+                  [0, 0, 0, 0, 0, 0, 0, 1]])
+
+ISWAP = np.array([[1, 0, 0, 0],
+                  [0, 0, 1j, 0],
+                  [0, 1j, 0, 0],
+                  [0, 0, 0, 1]])
+
+
+def PSWAP(phi):
+    return np.array([[1, 0, 0, 0],
+                     [0, 0, np.exp(1j * phi), 0],
+                     [0, np.exp(1j * phi), 0, 0],
+                     [0, 0, 0, 1]])
+
+
+# Utility gates for internal QVM use
+P0 = np.array([[1, 0], [0, 0]])
+
+P1 = np.array([[0, 0], [0, 1]])
+
+
+# Specialized useful gates; not officially in standard gate set
+def BARENCO(alpha, phi, theta):
+    lower_unitary = np.array([
+        [np.exp(1j * phi) * np.cos(theta), -1j * np.exp(1j * (alpha - phi)) * np.sin(theta)],
+        [-1j * np.exp(1j * (alpha + phi)) * np.sin(theta), np.exp(1j * alpha) * np.cos(theta)]])
+    return np.kron(P0, np.eye(2)) + np.kron(P1, lower_unitary)
+
+
+QUANTUM_GATES = {
+    'I': I,
+    'X': X,
+    'Y': Y,
+    'Z': Z,
+    'H': H,
+    'S': S,
+    'T': T,
+    'PHASE': PHASE,
+    'RX': RX,
+    'RY': RY,
+    'RZ': RZ,
+    'CNOT': CNOT,
+    'CCNOT': CCNOT,
+    'CPHASE00': CPHASE00,
+    'CPHASE01': CPHASE01,
+    'CPHASE10': CPHASE10,
+    'CPHASE': CPHASE,
+    'SWAP': SWAP,
+    'CSWAP': CSWAP,
+    'ISWAP': ISWAP,
+    'PSWAP': PSWAP,
+    'BARENCO': BARENCO,
+    'CZ': CZ
+}
+
+
+def relaxation_operators(p):
+    """
+    Return the amplitude damping Kraus operators
+    """
+    k0 = np.array([[1.0, 0.0], [0.0, np.sqrt(1 - p)]])
+    k1 = np.array([[0.0, np.sqrt(p)], [0.0, 0.0]])
+    return k0, k1
+
+
+def dephasing_operators(p):
+    """
+    Return the phase damping Kraus operators
+    """
+    k0 = np.eye(2) * np.sqrt(1 - p / 2)
+    k1 = np.sqrt(p / 2) * Z
+    return k0, k1
+
+
+def depolarizing_operators(p):
+    """
+    Return the phase damping Kraus operators
+    """
+    k0 = np.sqrt(1.0 - p) * I
+    k1 = np.sqrt(p / 3.0) * X
+    k2 = np.sqrt(p / 3.0) * Y
+    k3 = np.sqrt(p / 3.0) * Z
+    return k0, k1, k2, k3
+
+
+def phase_flip_operators(p):
+    """
+    Return the phase flip kraus operators
+    """
+    k0 = np.sqrt(1 - p) * I
+    k1 = np.sqrt(p) * Z
+    return k0, k1
+
+
+def bit_flip_operators(p):
+    """
+    Return the phase flip kraus operators
+    """
+    k0 = np.sqrt(1 - p) * I
+    k1 = np.sqrt(p) * X
+    return k0, k1
+
+
+def bitphase_flip_operators(p):
+    """
+    Return the bitphase flip kraus operators
+    """
+    k0 = np.sqrt(1 - p) * I
+    k1 = np.sqrt(p) * Y
+    return k0, k1
+
+
+KRAUS_OPS = {
+    'relaxation': relaxation_operators,
+    'dephasing': dephasing_operators,
+    'depolarizing': depolarizing_operators,
+    'phase_flip': phase_flip_operators,
+    'bit_flip': bit_flip_operators,
+    'bitphase_flip': bitphase_flip_operators,
+}
+
+__all__ = list(QUANTUM_GATES.keys()) + ['relaxation_operators', 'dephasing_operators',
```

Comment:
```suggestion
__all__ = list(QUANTUM_GATES) + ['relaxation_operators', 'dephasing_operators',
```

Reply:
I'd prefer to leave as-is, if that's all right with you. It's more obvious that I'm using the stringy keys with `keys` than using the (imo weird) fact that iterating over dictionaries iterates over keys